core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -505,6 +502,39 @@ private PlanBuilder planFrom(QuerySpecification node)
                 new ValuesNode(idAllocator.getNextId(), 1));
     }
 
+    // TODO this could be optimized to plan only the window specification parts that are actually used
```

Comment:
We should verify that any subquery in a window clause that's not being referenced is eventually removed from the plan.

Reply:
See the previous comment.